src/integration.test.ts

```diff
@@ -150,15 +151,30 @@ describe("Upload and download end-to-end tests", () => {
   });
 
   it("Should get JSON file contents", async () => {
-    const json = { key: "testdownload" };
-
     // Upload the data to acquire its skylink
     const file = new File([JSON.stringify(json)], dataKey, { type: "application/json" });
     const { skylink } = await client.uploadFile(file);
 
-    const { data } = await client.getFileContent(skylink);
+    const { data, contentType } = await client.getFileContent(skylink);
+
+    expect(data).toEqual(expect.any(Object));
+    expect(data).toEqual(json);
+    expect(contentType).toEqual("application/json");
+  });
+
+  it("Should get file contents when content type is not specified", async () => {
+    // Upload the data to acquire its skylink. Don't specify a content type.
+
+    const file = new File([JSON.stringify(json)], dataKey);
+    const { skylink } = await client.uploadFile(file);
+    expect(skylink).not.toEqual("");
+
+    // Get file content and check returned values.
+
+    const { data, contentType } = await client.getFileContent(skylink);
 
     expect(data).toEqual(expect.any(Object));
     expect(data).toEqual(json);
+    expect(contentType).toEqual("application/octet-stream");
```

Comment:
What happens if we specify the file's name and it has an extension? The content type should be guessed correctly, right? Can we add a test for that as well?

Actually, if the content allows it, we should get some content type heuristics even without a filename. All that "will read up to X bytes and guess the content" magic. Doesn't that work or is this JSON too short?

Reply:
I don't know why it doesn't infer from the content, but I added a test case for inferring from filename